### PR TITLE
Move OTel exporter API to autoconfigure distro API

### PR DESCRIPTION
**Pull request by jeanbisutti**
This PR:
* Rename the azure-monitor-opentelemetry-exporter artifact to azure-monitor-opentelemetry-autoconfigure
* Use the com.azure.monitor.opentelemetry.autoconfigure package for the public API
* Rename the AzureMonitorExporter class to AzureMonitorAutoconfigure
* Rename the AzureMonitorExporterOptions class to AzureMonitorAutoconfigureOptions

### CODE REVIEW

**Comment by jeanbisutti** [comment]
https://github.com/Azure/azure-sdk-for-java/blob/jean/otel-exporter-otel-autoconfigure/sdk/monitor/azure-monitor-opentelemetry-autoconfigure/assets.json needs to be updated to make the proxy tests work again.

**Comment by jeanbisutti** [comment]
CI pipelines may need to be updated:
*  https://github.com/Azure/azure-sdk-for-java/blob/jean/distro-api/sdk/monitor/ci.yml
* https://github.com/Azure/azure-sdk-for-java/blob/jean/distro-api/sdk/monitor/azure-monitor-opentelemetry-autoconfigure/tests.yml

**Comment by jeanbisutti** [comment]
CI pipelines may need to be updated:
* https://github.com/Azure/azure-sdk-for-java/blob/jean/otel-exporter-otel-autoconfigure/sdk/monitor/ci.yml
* https://github.com/Azure/azure-sdk-for-java/blob/jean/otel-exporter-otel-autoconfigure/sdk/monitor/azure-monitor-opentelemetry-autoconfigure/tests.yml

File: sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/AzureMonitorAutoconfigure.java [hunk 1]
```diff
@@ -18,9 +18,9 @@
 /**
  * Class to enable Azure Monitor for OpenTelemetry autoconfiguration.
  */
-public final class AzureMonitorExporter {
+public final class AzureMonitorAutoconfigure {
```
Comment (trask) [inline sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/AzureMonitorAutoconfigure.java:21]: let's follow OpenTelemetry's casing:

```suggestion
public final class AzureMonitorAutoConfigure {
```

File: eng/versioning/external_dependencies.txt [hunk 2]
```diff
@@ -50,6 +50,7 @@ com.microsoft.sqlserver:mssql-jdbc;10.2.3.jre8
 com.microsoft.azure:spring-data-cosmosdb;2.3.0
 com.microsoft.azure:azure-functions-maven-plugin;1.30.0
 com.microsoft.azure.functions:azure-functions-java-library;2.2.0
+com.azure:azure-monitor-opentelemetry-exporter;1.0.0-beta.31
```
[the remark this answers is not in the document]
Comment (JimSuplizio) [reply eng/versioning/external_dependencies.txt:53]: @saragluna is this true? Spring's hacked together pipeline code doesn't support what the engineering system does? This isn't okay.
@jeanbisutti if this is only a spring dependency only has to exist until com.azure:azure-monitor-opentelemetry-autoconfigure is released then I want an issue filed, assigned to you, to remove this external dependency and update the spring dependency com.azure:azure-monitor-opentelemetry-autoconfigure when it's released.

**Review by JimSuplizio — CHANGES_REQUESTED** [review]

**Comment by jeanbisutti** [comment]
> @jeanbisutti if this is only a spring dependency only has to exist until com.azure:azure-monitor-opentelemetry-autoconfigure is released then I want an issue filed, assigned to you, to remove this external dependency and update the spring dependency com.azure:azure-monitor-opentelemetry-autoconfigure when it's released.

@JimSuplizio Yes, please assign an issue to me

**Comment by jeanbisutti** [comment]
> @saragluna is this true? Spring's hacked together pipeline code doesn't support what the engineering system does? This isn't okay.
@jeanbisutti if this is only a spring dependency only has to exist until com.azure:azure-monitor-opentelemetry-autoconfigure is released then I want an issue filed, assigned to you, to remove this external dependency and update the spring dependency com.azure:azure-monitor-opentelemetry-autoconfigure when it's released.

@JimSuplizio We have already discussed about this in another place about the about the java - spring-monitor - tests pipeline

File: eng/versioning/external_dependencies.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (saragluna) [reply eng/versioning/external_dependencies.txt:53]: Sorry for the delayed response, but I don't remember we have done such changes to the pipelines.

**Comment by jeanbisutti** [comment]
Closed in favor of #42713

**Pull request closed without merging**